### DISCUSSION TITLE
Set Java version when running codeql tests

### DISCUSSION THREAD

**Discussion by ap00rv** [github/codeql#10391] [General]
Hi, I am trying to run a unit test locally and the test file uses Java Records, which are only supported from Java version 16+. 

When running `codeql test run` command, CodeQL by default uses Java version 8 to create database from test file. How can I make codeql use a different java version when running tests ? 

The javac error is like below : 

```
[2022-09-12 16:32:47] [javac-output-20649] warning: [options] bootstrap class path not set in conjunction with -source 8
[2022-09-12 16:32:47] [javac-output-20649] <redacted>.java:6: error: records are not supported in -source 8
[2022-09-12 16:32:47] [javac-output-20649] public record <redacted>(
[2022-09-12 16:32:47] [javac-output-20649]        ^
[2022-09-12 16:32:47] [javac-output-20649]   (use -source 16 or higher to enable records)


```

**Comment by aibaars** [comment]
You could try an [options](https://github.com/github/codeql/tree/main/java/ql/test/library-tests/types/record-classes/options) file as used by the test in [java/ql/test/library-tests/types/record-classes](https://github.com/github/codeql/tree/main/java/ql/test/library-tests/types/record-classes) . 

```
//semmle-extractor-options: --javac-args -source 16 -target 16
```

**Reply by ap00rv** [reply]
Thank you ! This worked.

**Comment by vorburger** [comment]
An alternative is to use an `actions/setup-java`, as explained e.g. in https://github.com/github/codeql-action/issues/825 or https://github.com/github/codeql-action/issues/97 or https://github.com/orgs/community/discussions/22425. (I've [used this here](https://github.com/vorburger/MariaDB4j/pull/691) and can confirm it works.)
